index.js

```diff
@@ -24,7 +24,18 @@ module.exports = function (settings) {
     }
   }
   apex.settings = settings
-  apex.domain = settings.domain
+  if (settings.baseUrl !== undefined) {
+    apex.baseUrl = settings.baseUrl
+    const url = new URL(apex.baseUrl)
+    apex.domain = url.hostname
+    if (url.port !== '') {
+      apex.domain += ':' + url.port
+    }
```

Comment:
`url.host` will give you what you want here for `apex.domain`: the hostname plus, if specified, the port

https://developer.mozilla.org/en-US/docs/Web/API/URL#instance_properties

```suggestion
    apex.domain = url.host
```